### ISSUE TITLE
Files in a zip file of a downloaded directory are read only on an Ubuntu Client

### ISSUE THREAD

**Issue by martinsauter** [owncloud/core#12505]
### Steps to reproduce
1. Have a directory on the OC server with several files in it.
2. Use Firefox to download the directory via the Web interface (as a ZIP file)
3. Unpack the ZIP file on Ubuntu 12.04 or 14.04 with the default Archive Manager (version 3.10.2.1)
4. After unpacking the files they are read-only

### Expected behaviour
The files should have read and write permissions after unpacking. This seems to be an Owncloud issue as the Archive Manager properly unpacks other archives and files with read and write permissions 

### Actual behaviour
All unpacked files are read only.

### Server configuration
**Operating system**: Ubuntu Server 12.04

**Web server:** Apache

**Database:** sqlite

**PHP version:** as supplied by Ubuntu 1204

**ownCloud version:** (see ownCloud admin page)

**Updated from an older ownCloud or fresh install:** 7.0.3 (stable)

**List of activated apps:** default ones

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => 'oc79dc19aab8',
  'passwordsalt' => 'xxxxxxxxxxx',
  'datadirectory' => 'xxxxxxxxx',
  'dbtype' => 'sqlite3',
  'version' => '7.0.3.4',
  'installed' => true,
  'loglevel' => '0',
  'maxZipInputSize' => 838860800,
  'allowZipDownload' => true,
  'enable_previews' => true,
  'theme' => '',
  'maintenance' => false,
  'mail_from_address' => 'owncloud',
  'mail_smtpmode' => 'smtp',
  'mail_domain' => 'xxxxx.com',
  'mail_smtpsecure' => 'tls',
  'mail_smtpauth' => true,
  'mail_smtpauthtype' => 'PLAIN',
  'mail_smtphost' => 'xxxxx',
  'mail_smtpport' => 'xxx',
  'mail_smtpname' => 'xxxxxxxxxxx',
  'mail_smtppassword' => 'xxxxxxxxxxxxxx',
);

```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

### Client configuration
**Browser:** Firefox Linux latest version

**Operating system:** Linux Ubuntu 14.04

#### Web server error log
```
None generated for the issue
```

#### ownCloud log (data/owncloud.log)
```
none generated for the issue
```

#### Browser log
```
NA
```

**Comment by martinsauter** [comment]
Still happening in Owncloud 8. Manually applying the fix suggested by McNetic above fixes the issue. Please consider merging the fix into the next maintenance release. Thanks!

**Comment by McNetic** [comment]
The fix is not merged because of other issues with ZipStreamer.

**Comment by PVince81** [comment]
@McNetic any update on this one ?

**Comment by McNetic** [comment]
Afaics, it was just agreed to update Zipstreamer in Owncloud to have all fixes from the last months (including this one) in #10001. I'm currently preparing a release (might take some days still, if I don't find time).

**Comment by martinsauter** [comment]
Cool, thanks, looking forward to it!

**Comment by martinsauter** [comment]
Still present in OC 8.1.1. I can see it's set to milestone "8.2". Would be really great if it finally made it into a release software after 9+ months...

**Comment by McNetic** [comment]
Well, I prepared the ZipStreamer release, so it's up to the owncloud developers to integrate it (see also #10001).

**Comment by McNetic** [comment]
This is fixed in master as of 5db186e (see also #19018).
